src/endless/EndlessUsbToolDlg.cpp

```diff
@@ -5531,7 +5531,17 @@ bool CEndlessUsbToolDlg::UnpackBootComponents(const CString &bootFilesZipPath, c
         return false;
     }
 
-    IFFALSE_RETURN_VALUE(UnpackZip(CComBSTR(bootFilesZipPath), CComBSTR(bootFilesPath)), "Error unpacking archive to local folder.", false);
+    // Rather than extracting the boot.zip's contents directly to a temporary
+    // directory, we first copy boot.zip to that temporary directory, then
+    // extract it. I know this seems pointless, but in fact it seems to work
+    // around an apparent Windows bug where the zip file is not properly
+    // unpacked if it's on a DVD, even though the exact same zip file is
+    // extracted properly when on any other medium.
+    // https://stackoverflow.com/q/47875272/173314
+    CString bootFilesZipCopyPath = bootFilesPath + CSTRING_GET_LAST(bootFilesZipPath, '\\');
```

Comment:
I suppose we could guard this to happen only in the case where `bootFilesPath.Left(2) != bootFilesZipPath.Left(2)`, ie when the zip file is on a different filesystem.